helix-core/src/test/resources/TestRecoveryLoadBalance.MasterSlave.json

```diff
@@ -253,5 +253,115 @@
         }
       }
     }
+  },
+  {
+    "statemodel": "MasterSlave",
+    "errorOrRecoveryPartitionThresholdForLoadBalance": 0,
+    "loadBalanceThrottle": "0",
+    "minActiveReplica": "2",
+    "inputs": {
+      "resource_0_0": {
+        "currentStates": {
+          "localhost_0": "MASTER",
+          "localhost_1": "SLAVE",
+          "localhost_2": "OFFLINE",
+          "localhost_3": "OFFLINE"
+        },
+        "bestPossibleStates": {
+          "localhost_0": "MASTER",
+          "localhost_1": "SLAVE",
+          "localhost_2": "SLAVE",
+          "localhost_3": "DROPPED"
+        },
+        "expectedStates": {
+          "localhost_0": "MASTER",
+          "localhost_1": "SLAVE",
+          "localhost_2": "OFFLINE",
+          "localhost_3": "OFFLINE"
+        }
+      },
+      "resource_0_1": {
+        "currentStates": {
+          "localhost_0": "SLAVE",
+          "localhost_1": "SLAVE",
+          "localhost_2": "OFFLINE"
+        },
+        "bestPossibleStates": {
+          "localhost_0": "MASTER",
+          "localhost_1": "SLAVE",
+          "localhost_2": "SLAVE"
+        },
+        "expectedStates": {
+          "localhost_0": "MASTER",
```

Comment:
Question: This is a case that we need to fix, correct? So we should expect this case to fail once we completely fix this?

This is a recovery balance due to the SLAVE->MASTER state transition, but OFFLINE->SLAVE is technically load balance which should be throttled?

Reply:
Right. But we need to add instance level throttling for this partition. Once we have the complete per replica level throttling, we should have other test cases to ensure that.